tests/python_package_test/test_engine.py

```diff
@@ -966,6 +966,26 @@ def test_continue_train_multiclass():
     assert evals_result['valid_0']['multi_logloss'][-1] == pytest.approx(ret)
 
 
+def test_continue_train_different_feature_size():
+    np.random.seed(0)
+    train_X = np.random.randn(100, 10)
+    train_y = np.sum(train_X, axis=1)
+    train_data = lgb.Dataset(train_X, label=train_y)
+    params = {
+        "objective": "regression",
+        "num_trees": 10,
+        "num_leaves": 31,
+        "verbose": 2,
```

Comment:
If you're not parsing output log, use `-1` to not overflood CI logs.
```suggestion
        "verbose": -1,
```